addons/account/models/mail_message.py

```diff
@@ -167,16 +204,17 @@ def _except_audit_log(self):
             # Invoicing/Customers menu)
             has_related_move = self.env['account.move'].sudo().search_count([
                 ('partner_id', 'in', partner_message.account_audit_log_partner_id.ids),
-                ('company_id.check_account_audit_trail', '=', True),
             ], limit=1)
             if not has_related_move:
                 to_check -= partner_message
         for message in to_check:
-            if message.account_audit_log_activated and not (
-                message.account_audit_log_move_id
-                and not message.account_audit_log_move_id.posted_before
-            ):
-                raise UserError(self.env._("You cannot remove parts of the audit trail."))
+            if message.account_audit_log_move_id and message.account_audit_log_move_id.state == 'draft':
+                return
+            # When a company has no move line yet, they can delete every log, as none is linked to a journal Item
+            # But we always want to know when was the last time the setting was set to True, so we need an additional check
+            setting_toggling = message.account_audit_log_company_id and "Restrictive Audit Trail" in message.account_audit_log_preview
```

Comment:
What about companies not using english?
Don't rely on display strings to manage business logic.